zssh/zscp/root.go

```diff
@@ -105,26 +122,93 @@ var (
 				Identity:       targetIdentity,
 				AppData:        nil,
 			}
-			svc, err := ctx.DialWithOptions(ExpectedServiceAndExeName, dialOptions)
+			svc, err := ctx.DialWithOptions(serviceName, dialOptions)
+			defer func() { _ = svc.Close() }()
 			if err != nil {
-				logrus.Fatal(fmt.Sprintf("error when dialing service name %s. %v", ExpectedServiceAndExeName, err))
+				logrus.Fatal(fmt.Sprintf("error when dialing service name %s. %v", serviceName, err))
 			}
 			factory := zsshlib.NewSshConfigFactoryImpl(username, SshKeyPath)
+			config := factory.Config()
+			sshConn, err := zsshlib.Dial(config, svc)
+			if err != nil {
+				logrus.Fatal(err, "error dialing SSH Conn")
+			}
+			client, err := sftp.NewClient(sshConn)
+			if err != nil {
+				logrus.Fatal(err, "error creating sftp client")
+			}
+			defer func() { _ = client.Close() }()
 
 			if isCopyToRemote {
-				zsshlib.SendFile(factory, localFilePath, remoteFilePath, svc)
+				if recursive {
+					baseDir := filepath.Base(localFilePath)
+					err := filepath.WalkDir(localFilePath, func(path string, info fs.DirEntry, err error) error {
+						remotePath := filepath.Join(remoteFilePath, baseDir, after(path, baseDir))
```

Comment:
what is `after` used for here? my initial reaction was "why is this here". it doesn't seem like it's needed?

Reply:
returns the end of path after the base directory. Needed a slick way to make sure all the target directory path names didn't pile onto each other.